crates/shadowsocks-service/src/acl/mod.rs

```diff
@@ -397,7 +398,11 @@ impl AccessControl {
             Address::SocketAddress(ref addr) => !self.check_ip_in_proxy_list(&addr.ip()),
             // Resolve hostname and check the list
             Address::DomainNameAddress(ref host, port) => {
-                if let Some(value) = self.check_host_in_proxy_list(host) {
+                // FIXME: Maybe return some error?
+                let is_matched = idna::domain_to_ascii(host)
+                    .map(|host| self.check_host_in_proxy_list(&host))
+                    .unwrap_or(None);
```

Comment:
I didn't put it in `check_host_matched` because it will lead to double conversion for both blacklist and whitelist here:
https://github.com/shadowsocks/shadowsocks-rust/blob/731e613a60364712b45f4de23db38e1df23ee220/crates/shadowsocks-service/src/acl/mod.rs#L347-L357

Reply:
Oh, that's a problem. So we should just put it in `check_host_in_proxy_list` and add the same logic in `check_outbound_blocked` for `self.outbound_block.check_host_matched`

Reply:
Rename this function to `check_idna_host_in_proxy_list` and make a new `check_host_in_proxy_list`.